### PR TITLE
DS-303 Button Twig update

### PR DESCRIPTION
**Pull request by cjwhitedev**
## Jira

[https://pegadigitalit.atlassian.net/browse/DS-303](https://pegadigitalit.atlassian.net/browse/DS-303)

## Summary

Update Button Twig conventions.

## Details

This updates Button to use `{% set this = init(schema) %}`. Also, we remove the `base_class` variable, switch from double to single quotes, and update any other Twig standards that were out of date.

## How to test

- Review files changed
- Visually compare Button demos and other components that use Button with demos on master

> Note: you'll see in Jest snapshots that default prop values are no longer output to `<bolt-button>`.

### CODE REVIEW

**Review by danielamorse — COMMENTED** [review]
@cjwhitedev nice job, this is a very tricky one. Button has a lot of gotchas. We're pretty close.

I'm in the process of addressing these inline comments myself cause I was already in the weeds debugging this stuff. I will finish this up on Tuesday.

File: packages/components/bolt-button/src/button.scss [hunk 1]
```diff
@@ -216,11 +216,13 @@ bolt-button[width='full@small'],
  * Border Radius
  */
 @each $radius-name, $radius-value in $bolt-border-radius-values {
-  .c-bolt-button--border-radius-#{$radius-name},
-  .c-bolt-button--border-radius-#{$radius-name}:before,
-  .c-bolt-button--border-radius-#{$radius-name}:after,
-  bolt-button[border-radius='#{$radius-name}'] {
-    border-radius: $radius-value;
+  @if $radius-name == 'full' {
```
Comment (danielamorse) [inline packages/components/bolt-button/src/button.scss:219]: @mikemai2awesome wanted to point out this edit to you. I noticed we were outputting unused radius values here.

**Review by danielamorse — APPROVED** [review]

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
@cjwhitedev Just one small update needed, the doc page still says "Style", we need to update it to "Hierarchy".

File: packages/components/bolt-button/src/button.scss (hunk 1 above)
Comment (mikemai2awesome) [reply packages/components/bolt-button/src/button.scss:219]: Yup all good.

**Comment by danielamorse** [comment]
> @cjwhitedev Just one small update needed, the doc page still says "Style", we need to update it to "Hierarchy".

@mikemai2awesome this is the change you requested right? https://github.com/boltdesignsystem/bolt/pull/2103/commits/4ccfb9123bc59a61474f129b89fec5e30559dba9

**Review by mikemai2awesome — APPROVED** [review]

**Pull request merged**